test/forward_backward_compatibility/check_forward_backward_compatibility.py

```diff
@@ -139,6 +137,42 @@ def allow_listed(schema):
     ("dist_c10d", datetime.date(2099, 9, 17)),
 ]
 
+def has_valid_upgraders(schema, version_map):
+    # we want to parse through the map to find if
+    # the schema has valid upgraders. Since the
+    # version map has entry for each overload
+    # we need to do some ugly parsing.
+
+    # the name of the operator
+    schema_name = schema.name
+
+    # find if there are entries for this op
+    possible_overloads = []
+    possible_schemas = []
+    for key, upgrader_entries in version_map.items():
```

Comment:
Shall we cache a "pre-processed" version_map? Its key should already be split for easier comparison, othewise we will be splitting them thousands of times for each test run

Reply:
sounds good